File: app/src/main/java/com/mrkostua/mathalarm/SharedPreferencesAlarmData.java
package com.mrkostua.mathalarm;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * @author Kostiantyn on 21.11.2017.
 */
//todo learn KOtlin companion object
public class SharedPreferencesAlarmData {
    private static SharedPreferences sharedPreferences;
    private static SharedPreferences.Editor editor;

    private static int hours;
    private static int minutes;
    private static String message;


    public SharedPreferencesAlarmData(Context context) {
        sharedPreferences = context.getSharedPreferences(ConstantValues.INSTANCE.getALARM_SHARED_PREFERENCE_NAME(), Context.MODE_PRIVATE);

    }

    public int getAlarmHours() {
        return sharedPreferences.getInt(ConstantValues.INSTANCE.getLAST_ALARM_HOURS(), -1);
    }

    public int getAlarmMinutes() {
        return sharedPreferences.getInt(ConstantValues.INSTANCE.getLAST_ALARM_MINUTES(), -1);
    }

    public String getAlarmTextMessage() {
        return sharedPreferences.getString(ConstantValues.INSTANCE.getSHARED_PREFERENCES_KEY_TEXT_MESSAGE(), ConstantValues.INSTANCE.getCUSTOM_ALARM_TEXT_MESSAGE());
    }

    // TODO: 08.12.2017 compile app whiten in Kotlin (some problems with generating method and null safety)
    public void saveLastAlarmData(AlarmObject alarmObject) {
        editor.putInt(ConstantValues.INSTANCE.getLAST_ALARM_HOURS(),
                alarmObject.getHours() != -1 ? alarmObject.getHours() : getAlarmHours());
        editor.putInt(ConstantValues.INSTANCE.getLAST_ALARM_MINUTES(),
                alarmObject.getMinutes() != -1 ? alarmObject.getMinutes() : getAlarmMinutes());
        editor.putString(ConstantValues.INSTANCE.getLAST_ALARM_MINUTES(),
                alarmObject.getTextMessage().equals(ConstantValues.INSTANCE.getCUSTOM_ALARM_TEXT_MESSAGE()) ? alarmObject.getTextMessage() : getAlarmTextMessage());

        editor.apply();
    }

}

File: app/src/main/java/com/mrkostua/mathalarm/SharedPreferencesAlarmData.kt
package com.mrkostua.mathalarm

import android.content.Context
import android.content.SharedPreferences

/**
 * @author Kostiantyn on 21.11.2017.
 */
class SharedPreferencesAlarmData(context: Context) {
    private val sharedPreferences = context.getSharedPreferences(ConstantValues.ALARM_SHARED_PREFERENCE_NAME, Context.MODE_PRIVATE)
    private val editor: SharedPreferences.Editor = sharedPreferences.edit()
    private val hours: Int = 0
    private val minutes: Int = 0
    private val message: String = ""
    private val ringtoneResId: Int = 0

    val alarmHours: Int
        get() = sharedPreferences.getInt(ConstantValues.LAST_ALARM_HOURS, -1)

    val alarmMinutes: Int
        get() = sharedPreferences.getInt(ConstantValues.LAST_ALARM_MINUTES, -1)

    val alarmTextMessage: String
        get() = sharedPreferences.getString(ConstantValues.SHARED_PREFERENCES_KEY_TEXT_MESSAGE, ConstantValues.CUSTOM_ALARM_TEXT_MESSAGE)

    val ringtoneResID: Int
        get() = sharedPreferences.getInt(ConstantValues.SHARED_PREFERENCES_KEY_TEXT_RINGTONE_RES_ID,ConstantValues.CUSTOM_ALARM_RINGTONE_ID )


    fun saveLastAlarmData(alarmObject: AlarmObject) {
        editor.putInt(ConstantValues.LAST_ALARM_HOURS,
                if (alarmObject.hours != -1) alarmObject.hours else alarmHours)
        editor.putInt(ConstantValues.LAST_ALARM_MINUTES,
                if (alarmObject.minutes != -1) alarmObject.minutes else alarmMinutes)
        editor.putString(ConstantValues.LAST_ALARM_MINUTES,
                if (alarmObject.textMessage == ConstantValues.CUSTOM_ALARM_TEXT_MESSAGE) alarmObject.textMessage else alarmTextMessage)

        editor.apply()
    }

}
